core/reactor_common.c

```diff
@@ -918,7 +946,8 @@ trigger_handle_t _lf_schedule(environment_t *env, trigger_t* trigger, interval_t
     if (!trigger->is_timer) {
         delay += trigger->offset;
     }
-    interval_t intended_time = env->current_tag.time + delay;
+    tag_t intended_tag = (tag_t){.time = env->current_tag.time + delay, .microstep = 0};
```

Comment:
It looks to me like the `.microstep` field is always zero. In that case, why are we recording a tag instead of a time? And on line 1108, is the microstep that we are writing to the heap going to be correct if the delay was zero and the microstep is supposed to be nonzero?

Reply:
A few lines later, the microstep may get incremented (this happens in federated execution if you get a barrage of tardy messages... They pile up in superdense time).